src/middlewares/error.ts

```diff
@@ -0,0 +1,37 @@
+import { NextFunction, Request, Response } from 'express';
+import { ValidationError } from 'yup';
+
+export class NoRouteError extends Error {
+  constructor(message: string) {
+    super(message);
+    this.name = 'NoRouteError';
+  }
+}
+
+/* --------------------
+   we use a consistent 200 for all responses to mark a "successfull conversation"
```

Comment:
this could be a bit confusing?

there are expected errors and unexpected errors and I am ok for expected errors to be 200 (e.g. route a not supported asset) but unexpected error should be 4xx or 5xx (unexpected internal server error should be 5xx)

this ensures we have visibility on errors as we can check datadog for error rate and setup alers if needed

Reply:
sure let's use the standard error code